server/sonar-server/src/main/java/org/sonar/server/webhook/WebhookPayloadFactoryImpl.java

```diff
@@ -66,8 +66,12 @@ private void writeServer(JsonWriter writer) {
   }
 
   private static void writeDates(JsonWriter writer, ProjectAnalysis analysis, System2 system2) {
-    analysis.getAnalysisDate().ifPresent(date -> writer.propDateTime("analysedAt", date));
-    writer.propDateTime("changedAt", analysis.getAnalysisDate().orElseGet(() -> new Date(system2.now())));
+    analysis.getAnalysis().ifPresent(a -> writer.propDateTime("analysedAt", a.getDate()));
+    if (analysis.getAnalysis().isPresent()) {
+      writer.propDateTime("changedAt", analysis.getAnalysis().get().getDate());
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Call "Optional#isPresent()" before accessing the value. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS3655)

Reply:
FP

Reply:
I will update to your suggestion. But why it's not a false positive ? The test is made on line 70, did it relate to multiple threads that can change the value of analysis.getAnalysis ?

Reply:
it's not a FP because there is no reason, from an external of view, to assume that two successive calls to `analysis.getAnalysis()` will return the same object.
to fix just the FP (and not improve the code), you should use a variable with the result of `analysis.getAnalysis()`